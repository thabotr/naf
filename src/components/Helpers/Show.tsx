export function Show({
  component,
  If,
  ElseShow,
}: {
  component: React.ReactNode;
  If: boolean;
  ElseShow: React.ReactNode;
}): JSX.Element {
  return <>{If ? component : ElseShow}</>;
}
