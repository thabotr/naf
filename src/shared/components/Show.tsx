import React, {ReactNode} from 'react';

export default function ({
  component,
  If,
  ElseShow,
}: {
  component: ReactNode;
  If: boolean;
  ElseShow: ReactNode;
}) {
  return <>{If ? component ?? null : ElseShow ?? null}</>;
}
