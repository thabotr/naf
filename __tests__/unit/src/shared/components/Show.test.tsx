import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';
import Show from '../../../../../src/shared/components/Show';

describe('HorizontalView', () => {
  const childForTruthy = <Text>I am here because If is truthy</Text>;
  const childForFalsy = <Text>I am here because If is falsy</Text>;
  it("correctly renders the child component in the 'component' prop given a truthy 'If' prop", () => {
    const tree = renderer
      .create(
        <Show component={childForTruthy} If={true} ElseShow={childForFalsy} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("correctly renders the child component in the 'ElseShow' prop given a falsy 'If' prop", () => {
    const tree = renderer
      .create(
        <Show component={childForTruthy} If={false} ElseShow={childForFalsy} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
