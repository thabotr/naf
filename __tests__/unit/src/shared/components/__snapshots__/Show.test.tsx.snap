// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalView correctly renders the child component in the 'ElseShow' prop given a falsy 'If' prop 1`] = `
<Text>
  I am here because If is falsy
</Text>
`;

exports[`HorizontalView correctly renders the child component in the 'component' prop given a truthy 'If' prop 1`] = `
<Text>
  I am here because If is truthy
</Text>
`;
